### PR TITLE
Add expired items command and display on startup

### PR DESCRIPTION
**Pull request by cnivedit**
Define ItemMap equality
Create method to fetch expired items from the inventory
Create method to print expired items
Add JUnit tests for getExpiredItems method
Create "expired" command to display expired items in the inventory

### CODE REVIEW

File: src/main/java/seedu/pill/util/ItemMap.java [hunk 1]
```diff
@@ -244,6 +268,33 @@ public void listItems() {
...
+            for (Map.Entry<String, TreeSet<Item>> entry : expiredItems.items.entrySet()) {
+                TreeSet<Item> itemSet = entry.getValue();
+                for (Item item : itemSet) {
+                    System.out.println(index + ". " + item.toString());
+                    index++;
+                }
+            }
+            System.out.println();
+        }
+    }
```
Comment (yijiano) [inline src/main/java/seedu/pill/util/ItemMap.java:296]: Perhaps include throw statements here for edge cases?

File: src/test/java/seedu/pill/util/GetExpiredItemsTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package seedu.pill.util;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class GetExpiredItemsTest {
```
Comment (yijiano) [inline src/test/java/seedu/pill/util/GetExpiredItemsTest.java:10]: Thanks for covering both successful and failure cases, as well as other edge cases. Good test coverage.

File: src/main/java/seedu/pill/util/ItemMap.java [hunk 3]
```diff
@@ -309,6 +360,31 @@ public ItemMap findItem(String itemName) {
...
+        ItemMap expiredItems = new ItemMap();
+        LocalDate currDate = LocalDate.now();
+        for (Map.Entry<String, TreeSet<Item>> entry : items.entrySet()) {
+            TreeSet<Item> itemSet = entry.getValue();
+            for (Item item : itemSet) {
+                item.getExpiryDate().ifPresent(expiry -> {
+                    if (expiry.isBefore(currDate)) {
+                        expiredItems.addItemSilent(item);
+                    }
+                });
```
Comment (yijiano) [inline src/main/java/seedu/pill/util/ItemMap.java:382]: Can consider using filter and map instead of if statements.

**Review by yijiano — APPROVED** [review]
LGTM
